canonical-web-and-design/maas-ui
# feat(ui): build ScrollFade component

## Done

- Built `ScrollFade` component, which adds a fade gradient to the bottom of a div if it's not at the bottom-most scroll position.
- Decided not to build a generic hook just yet - maybe when we need to track element scroll position for another component?

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Open `StorageResources.tsx` and change line 54 so that at least 3 storage pools will render.
- Go to the Project tab of a LXD KVM and check that the storage pool section of the summary card has a scrollbar
- Check that the section has a fade gradient, then scroll down to the bottom and check that the scrollbar disappears
- The same should happen in the "Virtual functions" section of the Resources tab

## Fixes

Fixes #2432 

## Screenshot

![Screenshot_2021-04-12 LXD project default bolla MAAS](https://user-images.githubusercontent.com/25733845/114325688-c0e9ee00-9b74-11eb-8597-54a12e8ab6cb.png)
